Logic Fixes

cassidoxa opened the pull request:
This adds a pearl requirement for Ice Rod Cave in inverted, a flippers requirement for inverted East Dark World Pier to prevent a player from having to reach it from IP without flippers, and changes the East Dark World River Pier rules to allow Qirn jump in glitch logic.

cassidoxa committed:
Add pearl requirement for Ice Rod Cave entrance

tcprescott commented:
Context:  We ran into this issue during an inverted crossword entrance race.
[spoiler_z6GKmxOJve.txt](https://github.com/KevinCathcart/ALttPEntranceRandomizer/files/5252836/spoiler_z6GKmxOJve.txt)

cassidoxa committed:
Add flippers requirement for no glitches inverted E Dark World Pier
Remove flippers from E Dark World River Pier base rule and add them back
in no glitches rules. This puts qirn jump into glitches logic.

aerinon commented on Rules.py:
```diff
@@ -878,6 +878,8 @@ def no_glitches_rules(world, player):
         set_rule(world.get_entrance('Dark Lake Hylia Ledge Drop', player), lambda state: state.has('Flippers', player))
+        set_rule(world.get_entrance('East Dark World Pier', player), lambda state: state.has('Flippers', player))
+        set_rule(world.get_entrance('East Dark World River Pier', player), lambda state: state.has('Flippers', player))
 
```
This overrides the Pearl requirement from global rules. Use add_rule instead. (This is a general problem with the no_glitches_section though)

aerinon replied:
Actually this might be okay since the pearl requirement isn't set in inverted.

cassidoxa replied:
I think this is okay because the first branch (above highlighted section) uses add_rule for East Dark World River Pier and should only be run when global_rules is called. The highlighted branch is run when the world mode is inverted and global_rules isn't called. Correct me if I'm wrong though.
